components/GoalDetailCard/index.js

```diff
@@ -53,6 +53,20 @@ export default function GoalDetailCard({ goal, deleteGoal }) {
     });
   }
 
+  function formatTimelyDate(timely) {
+    const isDate = !isNaN(Date.parse(timely));
+
+    if (isDate) {
+      const date = new Date(timely);
+      const day = date.getDate().toString().padStart(2, "0");
+      const month = (date.getMonth() + 1).toString().padStart(2, "0");
+      const year = date.getFullYear();
+      return `${day}.${month}.${year}`;
```

Comment:
If you want to get the current date in that format you could also try the integrated `toLocaleDateString`like this ([MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toLocaleDateString?retiredLocale=de)):

```suggestion
      const date = new Date().toLocaleDateString('de-DE', { dateStyle: "medium" })
      return date;
```

Reply:
I decided to do this instead `import { format } from "date-fns";`